stdlib/bytes.mli

```diff
@@ -173,6 +173,22 @@ val mapi : (int -> char -> char) -> bytes -> bytes
     index (in increasing index order) and stores the resulting bytes
     in a new sequence that is returned as the result. *)
 
+val fold : (char -> 'a -> 'a) -> bytes -> 'a -> 'a
```

Comment:
Oh cool -- didn't know that. Well, the function is cleaner in recursive form -- take a look and see if you like it.